api/enki.py

```diff
@@ -0,0 +1,633 @@
+        response = self.responses.pop()
+        return HTTP._process_response(
+            url, response, kwargs.get('allowed_response_codes'),
+            kwargs.get('disallowed_response_codes')
+        )
+
+    def _simple_http_get(self, url, headers, *args, **kwargs):
+        """Simulate Representation.simple_http_get."""
+        response = self._request_with_timeout('GET', url, *args, **kwargs)
+        return response.status_code, response.headers, response.content
+
+class EnkiBibliographicCoverageProvider(BibliographicCoverageProvider):
+    #TODO
+    """Fill in bibliographic metadata for Enki records.
+
+    Currently this is only used by BibliographicRefreshScript. It's
+    not normally necessary because the Enki API combines
+    bibliographic and availability data.
+    """
+    def __init__(self, _db, metadata_replacement_policy=None, enki_api=None,
+                 input_identifier_types=None, input_identifiers=None, **kwargs):
+        """
+        :param input_identifier_types: Passed in by RunCoverageProviderScript, data sources to get coverage for.
+        :param input_identifiers: Passed in by RunCoverageProviderScript, specific identifiers to get coverage for.
+        """
+        self.parser = BibliographicParser()
+        super(EnkiBibliographicCoverageProvider, self).__init__(
+            _db, enki_api, DataSource.ENKI,
+            batch_size=25,
+            metadata_replacement_policy=metadata_replacement_policy,
+            **kwargs
+        )
+
+    def process_batch(self, identifiers):
+        identifier_strings = self.api.create_identifier_strings(identifiers)
+        response = self.api.availability(title_ids=identifier_strings)
+        seen_identifiers = set()
+        batch_results = []
+        for metadata, availability in self.parser.process_all(response.content, "//enki:title"):
+            identifier, is_new = metadata.primary_identifier.load(self._db)
+            if not identifier in identifiers:
+                # Enki told us about a book we didn't ask
+                # for. This shouldn't happen, but if it does we should
+                # do nothing further.
+                continue
+            seen_identifiers.add(identifier.identifier)
+            result = self.set_metadata(identifier, metadata)
+            if not isinstance(result, CoverageFailure):
+                result = self.handle_success(identifier)
+            batch_results.append(result)
+
+        # Create a CoverageFailure object for each original identifier
+        # not mentioned in the results.
+        for identifier_string in identifier_strings:
+            if identifier_string not in seen_identifiers:
+                identifier, ignore = Identifier.for_foreign_id(
+                    self._db, Identifier.ENKI_ID, identifier_string
+                )
+                result = CoverageFailure(
+                    identifier, "Book not in collection", data_source=self.output_source, transient=False
```

Comment:
This should be treated as a transient error, since the book might be absent from Enki temporarily.

I would say "Book not found in ENKI" rather than "Book not in collection", since "not in collection" sounds like the book isn't in the collection on the local side -- the problem here is that the book _is_ in the collection on the local side but not on the remote.